src/EditorFeatures/Core.Wpf/InlineHints/InlineHintsTagger.cs

```diff
@@ -71,7 +73,22 @@ internal sealed class InlineHintsTagger : ITagger<IntraTextAdornmentTag>, IDispo
             _formatMap = taggerProvider.ClassificationFormatMapService.GetClassificationFormatMap(textView);
             _hintClassification = taggerProvider.ClassificationTypeRegistryService.GetClassificationType(InlineHintsTag.TagId);
             _formatMap.ClassificationFormatMappingChanged += this.OnClassificationFormatMappingChanged;
-            _tagAggregator.TagsChanged += OnTagAggregatorTagsChanged;
+            _tagAggregator.BatchedTagsChanged += TagAggregator_BatchedTagsChanged;
+        }
+
+        private void TagAggregator_BatchedTagsChanged(object sender, BatchedTagsChangedEventArgs e)
+        {
+            _cacheSnapshot = null;
+            _cache.Clear();
+            var mappingSpans = e.Spans;
+            foreach (var item in mappingSpans)
+            {
+                var spans = item.GetSpans(_buffer);
+                foreach (var span in spans)
+                {
+                    TagsChanged?.Invoke(this, new SnapshotSpanEventArgs(span));
```

Comment:
get and assign TagsChanged to a local first, check it for null, and do not proceed if null.